### ISSUE TITLE
All day events start time may be incorrect (iOS only)

### ISSUE THREAD

**Issue by scorpionate** [builttoroam/device_calendar#337]
I noticed  **startTimeZone: ekEvent.timeZone?.identifier**, might be null(nil) for all-day events.
Because of that, in my flutter code after calling _retrieveEvents()_ I received an all-day event object with changed start time property.

Let's say my event is at 06.10.2021 and my timezone is +2.
I got an event object with 
- start time at 05.10.2021, 22:00 (timezone difference exactly)
- end time at 06.10.2021, 24:00

SwiftDeviceCalendarPlugin.dart
```
 private func createEventFromEkEvent(calendarId: String, ekEvent: EKEvent) -> Event {
        ...
        ...
        let event = Event(
            eventId: ekEvent.eventIdentifier,
            calendarId: calendarId,
            title: ekEvent.title ?? "New Event",
            description: ekEvent.notes,
            start: Int64(ekEvent.startDate.millisecondsSinceEpoch),
            end: Int64(ekEvent.endDate.millisecondsSinceEpoch),
            startTimeZone: ekEvent.timeZone?.identifier,
            allDay: ekEvent.isAllDay,
            attendees: attendees,
            location: ekEvent.location,
            url: ekEvent.url?.absoluteString,
            recurrenceRule: recurrenceRule,
            organizer: convertEkParticipantToAttendee(ekParticipant: ekEvent.organizer),
            reminders: reminders,
            availability: convertEkEventAvailability(ekEventAvailability: ekEvent.availability)
        )

        return event
    }
```

**Comment by thomassth** [comment]
Can you try again with the latest build in #385. I think things should be fixed but I don't have an iOS device to confirm.

**Comment by OleksandraFedotova** [comment]
there is a broken part here https://github.com/builttoroam/device_calendar/blob/develop/ios/Classes/SwiftDeviceCalendarPlugin.swift
You set endDate to startDate if AllDay is true 
check the 818 line 
    if (isAllDay) { ekEvent!.endDate = startDate }
            else {
                ekEvent!.endDate = endDate

**Comment by OleksandraFedotova** [comment]
I added PR https://github.com/builttoroam/device_calendar/pull/450 that fixes that problem

**Comment by henry2man** [comment]
Can we consider this issue as fixed?

**Comment by IVLIVS-III** [comment]
@henry2man yes, good catch. The PR was merged in the 4.3 release.
Thanks for the reminder.
